runtime/runtime/src/lib.rs

```diff
@@ -2407,11 +2408,8 @@ mod tests {
         store_update.commit().unwrap();
 
         let contract_code = ContractCode::new(wasm_code, None);
-        let key = get_contract_cache_key(
-            &contract_code,
-            VMKind::default(),
-            &apply_state.config.wasm_config,
-        );
+        let vm_kind = VMKind::for_protocol_version(apply_state.current_protocol_version);
+        let key = get_contract_cache_key(&contract_code, vm_kind, &apply_state.config.wasm_config);
```

Comment:
Maybe pass `apply_state.current_protocol_version` to `get_contract_cache_key` to make selection logic to belong there?